paddle/phi/kernels/kps/reduce_sum_kernel.cu

```diff
@@ -13,11 +13,60 @@
 // limitations under the License.
 
 #include "paddle/phi/kernels/reduce_sum_kernel.h"
+#include <climits>
+#include "paddle/phi/core/enforce.h"
 #include "paddle/phi/core/kernel_registry.h"
+#include "paddle/phi/kernels/funcs/eigen/common.h"
 #include "paddle/phi/kernels/gpu/reduce.h"
 
 namespace phi {
 
+template <typename T,
+          int EigenDimSize = 5,
+          int ReducedDimSize = 1,
+          bool ReduceAll = false>
+void ReduceSumEigen(const KPDevice& dev_ctx,
+                    const DenseTensor& x,
+                    bool reduce_all,
+                    const std::vector<int64_t>& dims,
+                    DataType out_dtype,
+                    DenseTensor* out,
+                    std::vector<int>* reduce_dims) {
+  // Resize Input Tensor
+  auto new_x = x;
+  int added_dims = EigenDimSize - x.dims().size();
+  std::vector<int64_t> new_dim(added_dims, 1);
```

Comment:
Change `new_dim` to be `new_x_dim`, which is more readable.

Change the type of `new_dim` from `std::vector<int64_t>` to be `std::array<int64_t, EigenDimSize>`.